src/webgpu/api/operation/buffers/map_ArrayBuffer.spec.ts

```diff
@@ -38,42 +38,43 @@ g.test('postMessage')
     );
 
     await buf.mapAsync(GPUMapMode[mapMode]);
-    let ab1 = buf.getMappedRange();
+    const ab1 = buf.getMappedRange();
     t.expect(ab1.byteLength === kSize, 'ab1 should have the size of the buffer');
 
     const mc = new MessageChannel();
-    const ab2Promise = new Promise<ArrayBuffer>(resolve => {
-      mc.port2.onmessage = ev => resolve(ev.data);
+    const ab2Promise = new Promise<ArrayBuffer>((resolve, reject) => {
+      mc.port2.onmessage = ev => {
+        if (!transfer) {
+          resolve(ev.data);
+        } else {
+          reject(new Error('postMessage of transferred ArrayBuffer should fail'));
+        }
+      };
     });
 
-    mc.port1.postMessage(ab1, transfer ? [ab1] : undefined);
     if (transfer) {
-      t.expect(ab1.byteLength === 0, 'after postMessage, ab1 should be detached');
-      // Get the mapped range - again. So we can check that the data is not aliased
-      // with ab2. To do this, we need to unmap to reset the mapping state, and map
-      // again.
-      buf.unmap();
-      await buf.mapAsync(GPUMapMode[mapMode]);
-      ab1 = buf.getMappedRange();
-      t.expect(ab1.byteLength === kSize, 'ab1 should have the size of the buffer');
+      t.shouldThrow('TypeError', () => mc.port1.postMessage(ab1, [ab1]));
```

Comment:
another thought: it might be worth it to assert port2 doesn't receive the message in this case (would have to use some race-with-timeout thing)

Reply:
Done.